keps/sig-apps/0008-sidecarcontainer.md

```diff
@@ -0,0 +1,108 @@
+---
+kep-number: 0008
+title: Sidecar Containers
+authors:
+  - "@joseph-irving"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-apps
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-05-14
+last-updated: 2018-05-14
+status: provisional
+---
+
+# Sidecar Containers
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives-optional)
+
+## Summary
+
+To solve the issue of sidecars not being well supported we can create a new class of container in the Pod Spec called sidecarContainers which are handled like normal containers but are not considered when evaluating overall pod status. This will solve the problem of Jobs not completing/failing when a sidecar container never exits due to it having no awareness of the 'primary' container.
```

Comment:
I think we should be more specific here. There are many cases where we do want to consider sidecars when "evaluating overall pod status" since a failed sidecar container should change the pod status to "Failed" (and then be restarted). For this use case we're only interested in a pod where all sidecars are `Running` but every other pod has `Completed`.

Reply:
Yeah I agree. however there is also the case of when the main container has exited with an error but the sidecar is still running in which case the job won't fail. so I'd rephrase that as only interested in a pod where all sidecars are `Running` but every other container has `Completed/Error`.